modular/uploader/upload_task.go

```diff
@@ -184,3 +184,143 @@ func (u *UploadModular) QueryTasks(ctx context.Context, subKey coretask.TKey) (
 	uploadTasks, _ := taskqueue.ScanTQueueBySubKey(u.uploadQueue, subKey)
 	return uploadTasks, nil
 }
+
+func (u *UploadModular) PreResumableUploadObject(
+	ctx context.Context,
+	task coretask.ResumableUploadObjectTask) error {
+	if task == nil || task.GetObjectInfo() == nil || task.GetStorageParams() == nil {
+		log.CtxErrorw(ctx, "failed to pre upload object, task pointer dangling")
+		return ErrDanglingDownloadTask
+	}
+	if task.GetObjectInfo().GetObjectStatus() != storagetypes.OBJECT_STATUS_CREATED {
+		log.CtxErrorw(ctx, "failed to pre upload object, object not create")
+		return ErrNotCreatedState
+	}
+	if u.uploadQueue.Has(task.Key()) {
+		log.CtxErrorw(ctx, "failed to pre download object, task repeated")
+		return ErrRepeatedTask
+	}
+	if err := u.baseApp.GfSpClient().CreateResumableUploadObject(ctx, task); err != nil {
+		log.CtxErrorw(ctx, "failed to begin upload object task")
+		return err
+	}
+	return nil
+}
+
+func (u *UploadModular) HandleResumableUploadObjectTask(
```

Comment:
Do we support restarting the upload from the beginning if there is an error during file upload?

Reply:
But when the user use the restful api directly and he set the offset to zero, how to deal with it?